### DISCUSSION TITLE
While switching the branch , facing an error

### DISCUSSION THREAD

**Discussion by shweta13agg** [community/community#21382] [Other Feature Feedback, Questions, & Ideas]
<p>I’m remotely connected to the git project from the GIT Bash, while switching between the branches I’m facing the error</p>
<p>error: The following untracked working tree files would be overwritten by checkout:</p>
<p>        .gitconfig</p>
<p>Please move or remove them before you can switch branches.</p>
<p>Aborting</p>
<p>And when I’m doing the “git status”, it shows the local desktop changes not the project changes</p>
<p>        …/.eclipse/</p>
<p>        …/.gitconfig</p>
<p>        …/.jenkins/</p>
<p>        …/.m2/</p>
<p>        …/.node_repl_history</p>
<p>        …/.npmrc</p>
<p>        …/.oracle_jre_usage/</p>
<p>        …/.sonar/</p>
<p>        …/.ssh/</p>
<p>        …/.tooling/</p>
<p>        …/.viminfo</p>
<p>        …/.vscode/</p>
<p>        …/AppData/</p>
<p>        …/Citrix/</p>
<p>        …/Contacts/</p>
<p>        …/Desktop/</p>
<p>        …/Documents/</p>
<p>        …/Downloads/</p>
<p>        …/Favorites/</p>
<p>        …/Links/</p>
<p>        …/Music/</p>
<p>        …/Pictures/</p>
<p>        …/Saved Games/</p>
<p>        …/Searches/</p>
<p>        …/Videos/</p>
<p>        …/node_modules/</p>
<p>      </p>
<p>nothing added to commit but untracked files present (use “git add” to track)</p>
<p>What should be the approach to to resolve this issue?</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/shweta13agg">@shweta13agg</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>The error shows you the file that prevent you from checkout, that file is in the <strong>dirty</strong> form in that branch so you need to change his form by <strong>move it to staged</strong> area or <strong>delete</strong>  <strong>the changes</strong> made to that file.</p>
<p>To move to staged area you need to add that file and commit:</p>
<pre><code>git add .gitconfig
git commit -m 'Update (.gitconfig)"
</code></pre>
<p>To delete changes made to that file you need checkout that file:</p>
<pre><code>git checkout -- .gitconfig
</code></pre>
<p>The <strong>untracked</strong> files are the files that are not in the history yet. If you don’t want this files in the history you can add them to the <strong><em>.gitignore</em></strong> file, for example you don’t want the eclipse metadata you need to add the <strong>.eclipse</strong> directory into gitignore file:</p>
<pre><code>.eclipse/
</code></pre>
<p>And this resolve all your problem.</p>
<p><em>Suggestion: you’re trying to add your entire home to a git repository, be carefull with the dimension of what you push.</em></p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by shweta13agg** [comment]
<p>Thanks <a class="mention" href="/wabri">@wabri</a> it solve my issue.</p>
